lib/src/main/java/org/veupathdb/lib/jaxrs/raml/multipart/MultipartMessageBodyReader.kt

```diff
@@ -0,0 +1,238 @@
+package org.veupathdb.lib.jaxrs.raml.multipart
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import jakarta.ws.rs.BadRequestException
+import jakarta.ws.rs.InternalServerErrorException
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.MultivaluedMap
+import jakarta.ws.rs.ext.MessageBodyReader
+import org.apache.commons.fileupload.MultipartStream
+import org.veupathdb.lib.jaxrs.raml.multipart.utils.*
+import java.io.File
+import java.io.InputStream
+import java.lang.reflect.Type
+
+private const val DefaultFileName = "upload"
+
+private const val BufferSize = 8192
+
+class MultipartMessageBodyReader : MessageBodyReader<Any> {
+
+  /**
+   * Tests whether this `MessageBodyReader` should kick in for the given
+   * request.
+   *
+   * @return `true` if the input request is of type `multipart/form-data`,
+   * otherwise `false`.
+   */
+  override fun isReadable(
+    type: Class<*>,
+    genericType: Type,
+    annotations: Array<out Annotation>,
+    mediaType: MediaType,
+  ) = mediaType.isCompatible(MediaType.MULTIPART_FORM_DATA_TYPE)
+
+  override fun readFrom(
```

Comment:
The goal was more generally just to translate `multipart/form-data` requests into Jackson annotated POJOs.  This way the generated code can use `multipart/form-data` without regard to `File` fields.  I tried to encompass as many use cases as I could remember to avoid having to do any additional, or worse, per project logic around multipart endpoints.